server/bin/pbench-verify-backup-tarballs.py

```diff
@@ -0,0 +1,359 @@
+                open(indicator_file_ok, 'w') as f_ok,\
+                open(indicator_file_fail, 'w') as f_fail:
+            try:
+                for tar in tarlist:
+                    result_name = os.path.basename(tar)
+                    controller = os.path.basename(os.path.dirname(tar))
+                    md5 = "{}.md5".format(tar)
+                    f_list.write("{}\n".format(
+                        os.path.join(controller, result_name)))
+                    try:
+                        with open(md5) as f:
+                            md5_value = f.readline().split(" ")[0]
+                    except Exception:
+                        # Could not open the file
+                        nfailed_md5 += 1
+                        logger.exception(
+                            "Could not open the file {}".format(md5))
+                        continue
+                    md5_returned = md5sum(tar)
+                    if md5_value == md5_returned:
+                        f_ok.write("{}: {}\n".format(
+                            os.path.join(controller, result_name), "OK"))
+                    else:
+                        nfailed_md5 += 1
+                        f_fail.write("{}: {}\n".format(
+                            os.path.join(controller, result_name), "FAILED"))
+            except Exception:
+                logger.exception("Error processing list of matching tar balls")
+                nfailed_md5 = sys.maxsize
+    except Exception:
+        logger.exception(
+            "Could not open one of the temp files for writing {}".
+            format((indicator_file, indicator_file_ok, indicator_file_fail)))
+        nfailed_md5 = sys.maxsize
+    return nfailed_md5
+
+
+def report_failed_md5(backup_obj, tmpdir, report, logger):
+    fail_f = os.path.join(tmpdir, "{}.fail".format(backup_obj.list_name))
+    if os.path.exists(fail_f) and os.path.getsize(fail_f) > 0:
+        try:
+            with open(fail_f) as f:
+                failed_list = f.read()
+        except Exception:
+            # Could not open the file
+            logger.exception(
+                "Could not open the file {}".format(fail_f))
+        else:
+            report.write(
+                "ERROR: in {}: the calculated MD5 of the following entries "
+                "failed to match the stored MD5:\n{}".format(backup_obj.name, failed_list))
+
+
+def compare_with_s3_backup(s3_config_obj, backup_obj, tmpdir, report, logger):
+    if s3_config_obj is None:
+        return Status.FAIL
+
+    # Function to check integrity of results between archive/backup and s3
+    list_s3 = os.path.join(tmpdir, "list.s3")
+    try:
+        with open(list_s3, 'w') as f_list:
+            try:
+                s3_content_list = []
+                kwargs = {}
+                while True:
+                    resp = s3_config_obj.connector.list_objects(
+                        s3_config_obj.bucket_name, **kwargs)
```

Comment:
So, `list_objects()` needs the `kwargs` argument, otherwise the loop will not terminate. But you need to make sure that the underlying implementation of it handles it correctly. See the comment in `s3backup/__init__.py`.